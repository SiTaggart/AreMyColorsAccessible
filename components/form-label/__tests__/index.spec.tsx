/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import FormLabel from '..';

describe('FormLabel', (): void => {
  it('renders without crashing', (): void => {
    ReactDOM.render(
      <FormLabel htmlFor="input-id">children</FormLabel>,
      document.createElement('div')
    );
    const formLabelCmp = renderer
      .create(<FormLabel htmlFor="input-id">children</FormLabel>)
      .toJSON();
    expect(formLabelCmp).toMatchSnapshot();
  });

  it('should set create a large label style with isLarge', (): void => {
    const wrapper: ReactWrapper = mount(
      <FormLabel htmlFor="input-id" isLarge>
        children
      </FormLabel>
    );
    expect(wrapper.find('label').hasClass('form-label--large')).toEqual(true);
  });
});
