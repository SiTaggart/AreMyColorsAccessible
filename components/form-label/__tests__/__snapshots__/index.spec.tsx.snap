// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders without crashing 1`] = `
<label
  className="form-label"
  htmlFor="input-id"
>
  children
</label>
`;
